services/provider/server/server.go

```diff
@@ -2432,15 +2436,31 @@ func (s *OCSProviderServer) appendStorageClassKubeResources(
 		}
 	}
 	for i := range consumer.Spec.StorageClasses {
+		var storageClass *storagev1.StorageClass
 		storageClassName := consumer.Spec.StorageClasses[i].Name
 		scGen := scMap[storageClassName]
 		if scGen != nil {
-			sc := scGen()
-			sc.Name = storageClassName
-			kubeResources = append(kubeResources, sc)
+			storageClass = scGen()
+			storageClass.Name = storageClassName
 		} else {
-			//TODO: Day-2 storageClasses
-			klog.Warningf("encountered an unexpected storage class: %s", storageClassName)
+			var err error
+			storageClass, err = storageClassFromExisting(
+				ctx,
+				s.client,
+				storageClassName,
+				consumer,
+				consumerConfig,
+				drRbdStorageId,
+				drCephFsId,
```

Comment:
I am not sure where these are defined. But they should not have the `dr` prefix in the name